backends/github/labels/github-labels.js

```diff
@@ -0,0 +1,165 @@
+import GithubAPI from "../api/github-api.js";
+
+/**
+ * Backend for performing operations with labels on GitHub repositories.
+ * @category GitHub
+ */
+export default class GithubLabels extends GithubAPI {
+	// Do we need this at all?
+	static urlsKnown = [
+		{hostname: "api.github.com", pathname: "repos/:owner/:repo/labels"},
+		{hostname: "github.com", pathname: "/:owner/:repo/labels"},
+	];
+
+	static phrases = {
+		not_authenticated: "Please log in to perform actions with labels on GitHub.",
+		no_labels: "There are no labels to work with. Use the delete() method if you were to delete all existing labels in one go.",
+		success: (action) => `Labels ${action}d:`,
+		failure: (action) => `Labels failed to ${action}:`,
+	};
+
+	/**
+	 * Performs operations (create, update, delete) with labels.
+	 * @param {Array<any>} data Labels to create, update, or delete.
+	 * @returns {Promise<Object>} Promise that is resolved with an object based on the results of performed operations.
+	 */
+	async put (data, {ref = this.ref, force = false, skipDeletion = false} = {}) {
+		if (!this.isAuthenticated()) {
+			console.warn(this.constructor.phrase("not_authenticated"));
+			return null;
+		}
+
+		if (!data?.length && !force) {
+			// We don't want to delete all existing labels accidentally
+			console.warn(this.constructor.phrase("no_labels"));
```

Comment:
Hm. What if we actually *want* to delete all labels?

Reply:
Hm, not sure about this. Apps could be passing data to `store()` and it's annoying to have to check and use a different method if the array is empty. Not to mention it's a bit of a footgun as it's easy not to test for it. Think of a label management app that allows users to click x to easily delete a label. The user deletes labels one after another, but somehow mysteriously, the last one just won't get deleted. It's not great UX. And it doesn't protect you that much: overwriting dozens of labels with a single label is still just as problematic, and the check wouldn't prevent it. So, I don't think the cost-benefit is quite there.

That said, I think it's reasonable to go that route for `null` or `undefined`. But an empty array is pretty clear intent.

Reply:
Agreed